dpcpp/matrix/csr_kernels.dp.cpp

```diff
@@ -1625,6 +1625,27 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_CSR_ADVANCED_SPMV_KERNEL);
 
 
+template <typename ValueType, typename IndexType>
+void calculate_nonzeros_per_row_in_span(
+    std::shared_ptr<const DefaultExecutor> exec,
+    const matrix::Csr<ValueType, IndexType>* source, const span& row_span,
+    const span& col_span, Array<IndexType>* row_nnz) GKO_NOT_IMPLEMENTED;
```

Comment:
the row-parallel implementations from CUDA and HIP can be directly translated to implementations for DPCPP, can you add that here?